src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py

```diff
@@ -109,13 +109,20 @@ def evaluate(
         variables_map: VariablesMap,
         temporal_operational_settings: TemporalModel[List[SystemOperationalSettings]],
     ) -> EcalcModelResult:
+        """
+        Evaluating a consumer system that may be composed of both consumers and other consumer systems. It will default
+        to the last operational setting if all settings fails.
+
+        Todo: Regularity is only correct by chance, since it is given at the installation level only. We need to do
+            something about this: 'regularity=temporal_operational_settings.models[0].model[0].rates[0].regularity'
+        """
         is_valid = TimeSeriesBoolean(
             timesteps=variables_map.time_vector, values=[False] * len(variables_map.time_vector), unit=Unit.NONE
         )
         energy_usage = TimeSeriesRate(
             timesteps=variables_map.time_vector,
             values=[0] * len(variables_map.time_vector),  # TODO: Initialize energy usage to NaN not zero 4084
-            unit=Unit.STANDARD_CUBIC_METER_PER_DAY,
+            unit=Unit.NONE,
```

Comment:
so, this is because we do not know whether it is Sm3/d or MW, since it covers both el and fuel consumers? Either need to deduce this upon initialization of system or when collecting results from consumers?

Reply:
Yes, totally agree. See note in docstring

Reply:
yes, that is why I wanted it to be set directly, instead of doing it in 2 steps...and it seemd possible to me ...?

Reply:
It's only possible if there are consumers in the consumer system ... It may not be an option, but this is more defensive